src/main/java/dev/ebyrdeu/movie_jakarta/repository/MovieRepository.java

```diff
@@ -0,0 +1,44 @@
+package dev.ebyrdeu.movie_jakarta.repository;
+
+import dev.ebyrdeu.movie_jakarta.entity.Movie;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.transaction.Transactional;
+import jakarta.ws.rs.NotFoundException;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@ApplicationScoped
+public class MovieRepository {
+    @PersistenceContext(unitName = "mysql")
+    private EntityManager em;
+
+    @Transactional
+    public Movie saveMovie(Movie movie) {
+        em.persist(movie);
+        return movie;
+    }
+
+    public Optional<Movie> findById(UUID id) {
+        Movie movie = em.find(Movie.class, id);
+        return Optional.ofNullable(movie);
+    }
+
+    public List<Movie> findAll() {
```

Comment:
You can just return Movies instead of List<Movie>

```suggestion
    public Movies findAll() {
```

Reply:
If you don't have Movies use cherry pick on the Fredrick's commit about DTO, or merge with main we have small project so it won't(I hope) generate any conflicts

Reply:
> If you don't have Movies use cherry pick on the Fredrick's commit about DTO, or merge with main we have small project so it won't(I hope) generate any conflicts

I think both ways work. On one hand the repository should find entities and save entities. On the other hand data transferred inside the application should be done with DTOs. So as of now I think it works since we use this implementation to implement a service class that maps the list of Movie objects from repository, to a Movies object with mapped movieDtos in it. What do you think?

Reply:
It certainly works. My issue is that it just negates the purpose of the Movies DTO. We need some sort of consistency; either we should discard the Movies DTO and use List<Movie>, or do the opposite. 

Also, one thing I forgot to mention before - is that if we decide to use the Service class, we still need to create update method in the Repository. You can't just skip one method and added in another Class it's not what is Service for; it's a layer in between. So, if we're going to use the Service, both of them (Service and Repository) need to have full CRUD capabilities. Alternatively, we can simplify things and just use the update method in the Repository.

Reply:
We can leave it as it is since it doesn't cause any harm; it's just a concern of mine. However, Adam still needs to create update method regardless.

Reply:
> It certainly works. My issue is that it just negates the purpose of the Movies DTO. We need some sort of consistency; either we should discard the Movies DTO and use List, or do the opposite.
> 
> Also, one thing I forgot to mention before - is that if we decide to use the Service class, we still need to create update method in the Repository. You can't just skip one method and added in another Class it's not what is Service for; it's a layer in between. So, if we're going to use the Service, both of them (Service and Repository) need to have full CRUD capabilities. Alternatively, we can simplify things and just use the update method in the Repository.

I agree on the DTO.

If we do put update in repository we are going to have two methods in repository doing the same thing. saveMovie() will do the same as updateMovie(). Is that proper?

Reply:
Ah, I see the problem now. You are essentially confused about the naming convention in the Repository (which is why I prefer DAO). To clarify, no, `saveMovie` does not update anything.

If you examine closely the method I used here, you will notice `persist()`. The thing is, persist solely saves a new value to the database; it does not alter any existing value.

When we update in Hibernate, we use `merge()`. That is essentially our `update` method.

Although we have several more methods that do the same thing but have different names, we don't need to use them. If you're interested in reading about them, I can give you a [link](https://www.baeldung.com/hibernate-save-persist-update-merge-saveorupdate). But don't overthink it; in 99% situations, for creating data we use `persist`, and if we want to update data, we use `merge`.

In terms of what to name the method, simply using `update` is sufficient.

Reply:
In terms of how it would look if we just want to use the Repository, we can structure it somewhat like this. (Although with a service layer, it's going to look a bit different.)

```java
       @Transactional
    public Movie update(Movie movie) {
        var existingMovie = em.find(Movie.class, movie.getId());

        if (existingMovie == null) {
            throw new NotFoundException("Cannot find movie with id: " + movie.getId());
        }

        return em.merge(existingMovie);
    }


```